expr/func_groupbytags_test.go

```diff
@@ -0,0 +1,349 @@
+package expr
+
+import (
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/test"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+func getModel(name string, data []schema.Point) models.Series {
+	tags := make(map[string]string, 3)
+	tagSplits := strings.Split(name, ";")
+
+	tags["name"] = tagSplits[0]
+
+	for _, split := range tagSplits[1:] {
+		pair := strings.SplitN(split, "=", 2)
+		tags[pair[0]] = pair[1]
+	}
+	return models.Series{
+		Target:     name,
+		QueryPatt:  name,
+		Tags:       tags,
+		Datapoints: getCopy(data),
+	}
+}
+
+// Test error cases
+func TestNoTags(t *testing.T) {
+	in := []models.Series{
+		getModel("name1;tag1=val1", a),
+	}
+	f := NewGroupByTags()
+	gby := f.(*FuncGroupByTags)
+	gby.in = NewMock(in)
+	gby.aggregator = "sum"
+	gby.tags = []string{}
+
+	_, err := f.Exec(make(map[Req][]models.Series))
+	if err == nil {
+		t.Fatalf("case %q: err should not be nil but was", "TestnoTags")
+	}
+}
+
+func TestInvalidAggregator(t *testing.T) {
+	in := []models.Series{
+		getModel("name1;tag1=val1", a),
+	}
+	f := NewGroupByTags()
+	gby := f.(*FuncGroupByTags)
+	gby.in = NewMock(in)
+	gby.aggregator = "inv"
+	gby.tags = []string{}
+
+	_, err := f.Exec(make(map[Req][]models.Series))
+	if err == nil {
+		t.Fatalf("case %q: err should not be nil but was", "TestnoTags")
+	}
+}
+
```

Comment:
for the erroring cases, wouldn't it make sense to rely on testGroupByTags also? (just extend it to check for the error?)